syntax = "proto3";

package silenteight.agent.geo.v1.api;

import public "silenteight/agent/geo/v1/api/common.proto";
import "google/api/field_behavior.proto";

option java_package = "com.silenteight.proto.agent.geo.v1.api";
option java_outer_classname = "GeoLocationAgentApiProto";
option java_multiple_files = true;

service GeoLocationAgent {

  rpc CompareLocations (CompareLocationsRequest) returns (CompareLocationsResponse) {
  }

  rpc BatchCompareLocations (BatchCompareLocationsRequest) returns (BatchCompareLocationsResponse) {
  }
}

message BatchCompareLocationsRequest {
  repeated CompareLocationsRequest requests = 1 [(google.api.field_behavior) = REQUIRED];
}

message BatchCompareLocationsResponse {
  repeated CompareLocationsResponse responses = 1 [(google.api.field_behavior) = REQUIRED];
}

message CompareLocationsRequest {
  string alerted_party_location = 1 [(google.api.field_behavior) = REQUIRED];
  string watchlist_party_location = 2 [(google.api.field_behavior) = REQUIRED];
}

message CompareLocationsResponse {
  string solution = 1 [(google.api.field_behavior) = REQUIRED];
  GeoComparerReason reason = 2 [(google.api.field_behavior) = REQUIRED];
  IdentifiedLocations identified_inputs = 3 [(google.api.field_behavior) = REQUIRED];
  IdentifiedLocations skipped_inputs = 4 [(google.api.field_behavior) = OPTIONAL];
}

message IdentifiedLocations {
  repeated IdentifiedLocation alerted_party_identified_locations = 1 [(google.api.field_behavior) = REQUIRED];
  repeated IdentifiedLocation watchlist_party_identified_locations = 2 [(google.api.field_behavior) = REQUIRED];
}
